app/src/app/api/v0/city/[city]/inventory/[year]/progress/route.ts

```diff
@@ -3,118 +3,128 @@ import { Sector } from "@/models/Sector";
 import { SectorValue } from "@/models/SectorValue";
 import { apiHandler } from "@/util/api";
 import createHttpError from "http-errors";
+import { Session } from "next-auth";
 import { NextRequest, NextResponse } from "next/server";
 
-export const GET = apiHandler(async (_req: NextRequest, { params }) => {
-  const city = await db.models.City.findOne({ where: { locode: params.city } });
-  if (!city) {
-    throw new createHttpError.NotFound("City not found");
-  }
+export const GET = apiHandler(
+  async (
+    _req: NextRequest,
+    context: { session?: Session; params: Record<string, string> },
+  ) => {
+    const { params, session } = context;
+    const city = await db.models.City.findOne({
+      where: { locode: params.city },
+    });
+    if (!session) throw new createHttpError.Unauthorized("Unauthorized");
```

Comment:
Also, here I think you also have to check if the user ID from the session is allowed to access data from this city. This could also be part of the code in apiHandler imo (at least if there is a cityId in the route params).